render/render.js

```diff
@@ -489,18 +489,18 @@ module.exports = function($window) {
 	}
 
 	//lifecycle
-	function initLifecycle(source, vnode, hooks) {
-		if (typeof source.oninit === "function") source.oninit.call(vnode.state, vnode)
-		if (typeof source.oncreate === "function") hooks.push(source.oncreate.bind(vnode.state, vnode))
+	function initLifecycle(vnode, source, hooks) {
```

Comment:
Also, you'd have to cache it twice, the point of this change is to allow `oninit` to set `vnode.state`.